src/platform/i386/vga.c

```diff
@@ -0,0 +1,211 @@
+	 *  the memset, provided that your local variables if any, are
+	 *  unsigned short */
+	temp = (unsigned short *)dest;
+	for( ; count != 0; count--) *temp++ = val;
+	return dest;
+}
+
+/* Updates the hardware cursor: the little blinking line
+ *  on the screen under the last character pressed! */
+static void move_csr(void)
+{
+	unsigned pos = 0;
+
+	/* The equation for finding the index in a linear
+	 *  chunk of memory can be represented by:
+	 *  Index = [(y * width) + x] */
+	pos = csr_y * COLUMNS + csr_x;
+
+	/* This sends a command to indicies 14 and 15 in the
+	 *  CRT Control Register of the VGA controller. These
+	 *  are the high and low bytes of the index that show
+	 *  where the hardware cursor is to be 'blinking'. To
+	 *  learn more, you should look up some VGA specific
+	 *  programming documents. A great start to graphics:
+	 *  http://www.brackeen.com/home/vga */
+	outb(VGA_CTL_REG, 14);
+	outb(VGA_DATA_REG, pos >> 8);
+	outb(VGA_CTL_REG, 15);
+	outb(VGA_DATA_REG, pos);
+}
+
+/* Clears the line from current y position */
+static void cll(void)
+{
+	unsigned blank = 0;
+	/* Again, we need the 'short' that will be used to
+	 *  represent a space with color */
+	blank = SPACE | (attrib << 8);
+	/* Sets the entire screen to spaces in our current
+	 *  color */
+	memsetw ((unsigned short *)video + csr_y * COLUMNS + csr_x, blank, COLUMNS - csr_x);
+}
+
+
+/* Clears the screen */
+static void cls(void)
+{
+	unsigned blank = 0;
+	int i = 0;
+
+	/* Again, we need the 'short' that will be used to
+	 *  represent a space with color */
+	blank = SPACE | (attrib << 8);
+
+	/* Sets the entire screen to spaces in our current
+	 *  color */
+	for(i = 0; i < LINES; i++)
+		memsetw ((unsigned short *)video + i * COLUMNS, blank, COLUMNS);
+
+	/* Update out virtual cursor, and then move the
+	 *  hardware cursor */
+	csr_x = 0;
+	csr_y = 0;
+	move_csr();
+}
+
+/* Clear the screen and initialize VIDEO, XPOS and YPOS. */
+void vga_init (void)
+{
+	int i = 0;
+
+	video = (unsigned char *) VIDEO;
+
+	csr_x = 0;
+	csr_y = 0;
+	cls();
+	printk_register_handler(vga_puts);
+}
+
+/* Initialize VIDEO virtual address - High address. */
+void vga_high_init (void)
+{
+	video = chal_pa2va(VIDEO);
+}
+
+/* Put the character C on the screen. */
+static void putchar (int c)
+{
+	if (c == '\n' || c == '\r') {
+newline:
+		cll();
+		csr_x = 0;
+		csr_y++;
+		if (csr_y >= LINES)
+			/* rotation */
+			csr_y = 0;
+		return;
+	}
+
+	*(video + (csr_x + csr_y * COLUMNS) * 2) = c & 0xFF;
+	*(video + (csr_x + csr_y * COLUMNS) * 2 + 1) = attrib;
+
+	csr_x++;
+	if (csr_x >= COLUMNS)
+		goto newline;
+}
+
+void vga_puts(const char *s)
+{
+	puts((unsigned char *)s);
+}
+
+/* Uses the above routine to output a string... */
+static void puts(unsigned char *text)
+{
+	size_t i = 0;
+
+	cll();
+	for (i = 0; i < strnlen((const char*)text, STRLEN_MAX); i++) {
+		putchar(text[i]);
+	}
+	move_csr();
+}
+
+void keyboard_handler(struct pt_regs *regs)
```

Comment:
Just checked, only keyboard_handler() is in console.o and that is not disabled in Jacob's code but in my commit, console.o is disabled and replaced with vga.o and I've included this handler in vga.o, that should be fine right?